WcaOnRails/db/seeds/development/users.seeds.rb

```diff
@@ -13,37 +13,47 @@ def self.random_user
   end
 
   # Create board members
+  board_team = Team.find_by_slug('board-members')
   8.times do
-    FactoryGirl.create(:board_member)
+    FactoryGirl.create(:team_member, :board_member, team: board_team)
   end
 
   # Create senior delegates and their subordinate delegates
+  delegates_committee = Committee.find_by_slug(Committee::WCA_DELEGATES_COMMITTEE)
+  delegate_team = Team.where("slug = :slug", slug: 'delegate-testing-team').first_or_create(name: 'Delegate Testing Team', description: 'For testing', committee_id: delegates_committee.id)
   5.times do
-    senior_delegate = FactoryGirl.create(:senior_delegate)
+    FactoryGirl.create(:team_member, :senior_delegate, team: delegate_team)
     rand(10).times do
-      FactoryGirl.create([ :delegate, :candidate_delegate ].sample,
-                         senior_delegate: senior_delegate)
+      FactoryGirl.create(:team_member, [ :delegate, :candidate_delegate ].sample, team: delegate_team)
     end
   end
 
-  # Create some delegates without a senior delegate
-  5.times do
-    FactoryGirl.create([ :delegate, :candidate_delegate ].sample)
+  # Create a leader and members for the software WCA committee
+  software_team = Team.find_by_slug('software-team')
+  FactoryGirl.create(:team_member, :software_team_leader, team: software_team)
+  3.times do
+    FactoryGirl.create(:team_member, team: software_team)
   end
 
-  # Create members and leaders for every WCA team
-  UsersController.WCA_TEAMS.each do |team_friendly_id|
-    team_id = Team.find_by_friendly_id(team_friendly_id).id
+  # Create a leader and members for the results WCA committee
+  results_team = Team.find_by_slug('results-team')
+  FactoryGirl.create(:team_member, :results_team_leader, team: results_team)
+  3.times do
+    FactoryGirl.create(:team_member, team: results_team)
+  end
 
-    leader = FactoryGirl.create(:user)
-    FactoryGirl.create(:team_member, user_id: leader.id, team_id: team_id, team_leader: true)
-    # The team name isn't a valid email, but it's so much easier to type.
-    leader.update_column(:email, "#{team_friendly_id}_team")
+  # Create a leader and members for the regulations WCA committee
+  regulations_team = Team.find_by_slug('regulations-team')
+  FactoryGirl.create(:team_member, :regulations_team_leader, team: regulations_team)
+  3.times do
+    FactoryGirl.create(:team_member, team: regulations_team)
+  end
 
-    3.times do
-      member = FactoryGirl.create(:user)
-      FactoryGirl.create(:team_member, user_id: member.id, team_id: team_id)
-    end
+  # Create a leader and members for the disciplinary WCA committee
+  disciplinary_team = Team.find_by_slug('disciplinary-team')
+  FactoryGirl.create(:team_member, :disciplinary_team_leader, team: disciplinary_team)
+  3.times do
+    FactoryGirl.create(:team_member, team: disciplinary_team)
   end
 end
```

Comment:
Does it make sense to pass `team:` to the team_member factory? That's what the traits like `disciplinary_team_leader` takes care of.